src/pages/Staking/index.tsx

```diff
@@ -1,10 +1,747 @@
+      address,
+      monetaryLockingAmount
+    ],
+    genericFetcher<RewardAmountAndAPY>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(estimatedRewardAmountAndAPYError);
+
+  const {
+    isIdle: stakedAmountAndEndBlockIdle,
+    isLoading: stakedAmountAndEndBlockLoading,
+    data: stakedAmountAndEndBlock,
+    error: stakedAmountAndEndBlockError,
+    refetch: stakedAmountAndEndBlockRefetch
+  } = useQuery<StakedAmountAndEndBlock, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'getStakedBalance',
+      address
+    ],
+    genericFetcher<StakedAmountAndEndBlock>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(stakedAmountAndEndBlockError);
+
+  const initialStakeMutation = useMutation<void, Error, LockingAmountAndTime>(
+    (variables: LockingAmountAndTime) => {
+      if (currentBlockNumber === undefined) {
+        throw new Error('Something went wrong!');
+      }
+      const unlockHeight = currentBlockNumber + getLockBlocks(variables.time);
+
+      return window.bridge.interBtcApi.escrow.createLock(variables.amount, unlockHeight);
+    },
+    {
+      onSuccess: () => {
+        voteGovernanceTokenBalanceRefetch();
+        stakedAmountAndEndBlockRefetch();
+        rewardAmountAndAPYRefetch();
+        reset({
+          [LOCKING_AMOUNT]: '0.0',
+          [LOCK_TIME]: '0'
+        });
+      }
+    }
+  );
+
+  const moreStakeMutation = useMutation<void, Error, LockingAmountAndTime>(
+    (variables: LockingAmountAndTime) => {
+      return (async () => {
+        if (currentBlockNumber === undefined) {
+          throw new Error('Something went wrong!');
+        }
+        if ( // Increase amount and extend lock time
+          variables.time > 0 &&
+          variables.amount.gt(ZERO_GOVERNANCE_TOKEN_AMOUNT)
+        ) {
+          const unlockHeight = currentBlockNumber + getLockBlocks(variables.time);
+
+          const txs = [
+            window.bridge.interBtcApi.api.tx.escrow.increaseAmount(
+              variables.amount.toString(variables.amount.currency.rawBase)
+            ),
+            window.bridge.interBtcApi.api.tx.escrow.increaseUnlockHeight(unlockHeight)
+          ];
+          const batch = window.bridge.interBtcApi.api.tx.utility.batchAll(txs);
+          await DefaultTransactionAPI.sendLogged(
+            window.bridge.interBtcApi.api,
+            window.bridge.interBtcApi.account as AddressOrPair,
+            batch
+          );
+        } else if ( // Only increase amount
+          variables.time === 0 &&
+          variables.amount.gt(ZERO_GOVERNANCE_TOKEN_AMOUNT)
+        ) {
+          return await window.bridge.interBtcApi.escrow.increaseAmount(variables.amount);
+        } else if ( // Only extend lock time
+          variables.time > 0 &&
+          variables.amount.eq(ZERO_GOVERNANCE_TOKEN_AMOUNT)
+        ) {
+          const unlockHeight = currentBlockNumber + getLockBlocks(variables.time);
```

Comment:
Similar to above, I'm not sure exactly how the `endBlock` is stored right now but something like this should fix the bug:

```ts
const unlockHeight = LockingAmountAndTime.endBlock + getLockBlocks(variables.time);
```

Reply:
Done!